src/org/testKitGen/TestInfoParser.java

```diff
@@ -444,28 +444,42 @@ private boolean matchPlat(String fullSpec, String pr) {
 		}
 
 		if (prSplitOnDot[0].equals("arch") && (prSplitOnDot.length == 3)) {
-			String microArch = prSplitOnDot[2];
-			if (!microArch.equals(arg.getMicroArch())) {
-				return false;
+			String microArchVersion = prSplitOnDot[2];
+			String environmentMicroArch = arg.getMicroArch();
+			if (microArchVersion.endsWith("+")) {
+				int microArchVersionInt, environmentMicroArchVerInt;
+				try {
+					microArchVersionInt = extractVersionNumber(microArchVersion.substring(0, microArchVersion.length() - 1), "platformRequirement");
+					environmentMicroArchVerInt = extractVersionNumber(environmentMicroArch, "environmentMicroArch");
+				} catch (NumberFormatException e) {
+					System.err.println(e.getMessage());
+					return false;
+				}
+				return microArchVersionInt <= environmentMicroArchVerInt;
+			} else {
+				return environmentMicroArch.equals(microArchVersion);
```

Comment:
- Please use https://github.com/adoptium/TKG/pull/477/files#diff-1ce05e4d3d428c3c8430b54f2905415f30fd13919ff8269e99c127485d69093eR462-R499 as reference. For number comparison, please make a function if needed.
- We need to compare the first character as well. i.e., for arch.390.z15+, we need to ensure `z` matches.